### ISSUE TITLE
"Replacement has x rows, data has y rows" Error when using liwcalike() 

### ISSUE THREAD

**Issue by lin-jennifer** [kbenoit/quanteda.dictionaries#26]
I am trying to analyze the attached dataset using the `liwcalike()` command in the context of the Moral Foundations Dictionary provided with this package.

Here is my code:

```{r}
#Qualitative Analysis in R

load("data_dictionary_MFD.rda")
library(readtext)

#Run using the ANES 12 with feeling thermometers
like = readtext("like-dislike.csv", text_field = c("candlik_likewhatdpc", "candlik_dislwhatdpc", "candlik_likewhatrpc", "candlik_dislwhatrpc", "ptylik_lwhatdp", "ptylik_dwhatdp", "ptylik_lwhatrp", "ptylik_dwhatrp"))
summary(like)
na.omit(like)
#Create Corpus
require(quanteda)
library(quanteda.corpora)
data_like_dislike = corpus(like, docid_field = "caseid")

summary(data_like_dislike)

docvars(data_like_dislike, setdiff(names(docvars(data_like_dislike)), 
                            names(docvars(data_like_dislike)))) <- NULL

#Create DFM using MFD Dictionary
data_likedfm = dfm(data_like_dislike, dictionary = data_dictionary_MFD)
data_like = dfm_subset(data_likedfm, ntoken(data_likedfm) > 0)

#LIWCalike analysis - Load in the data dictionary package
library(quanteda.dictionaries)

MFD <- liwcalike(data_like_dislike, dictionary = data_dictionary_MFD)

head(MFD)
```

Yet, when I run this code, I receive the following error:

```{r}
> #LIWCalike analysis - Load in the data dictionary package
> library(quanteda.dictionaries)
> MFD <- liwcalike(data_like_dislike, dictionary = data_dictionary_MFD)
Error in `[[<-.data.frame`(`*tmp*`, "WPS", value = c(16, 26, 6, 6.33333333333333,  : 
  replacement has 5165 rows, data has 5914
> head(MFD)
Error in head(MFD) : object 'MFD' not found
```

Please advise me on how I can fix this error.

Thank you!

[like-dislike.xlsx](https://github.com/kbenoit/quanteda.dictionaries/files/3081622/like-dislike.xlsx)

**Comment by kbenoit** [comment]
Thanks @lin-jennifer, this appears to be a bug, we will try to fix it asap. 

You might study the first few lines to see a better way of working with your data. (Ignore the `download.files()` part, that's just for reproducibility.)

``` r
# load in the data and tidy it up

tfile <- tempfile()
download.file("https://github.com/kbenoit/quanteda.dictionaries/files/3081622/like-dislike.xlsx",
  destfile = tfile
)

like <- readxl::read_xlsx(tfile)
# sstack into one column
like2 <- tidyr::gather(like, "variable", "text", -caseid)
# remove missing
like2 <- na.omit(like2)

# quanteda analysis

library("quanteda")
## Package version: 1.4.4
## Parallel computing: 2 of 12 threads used.
## See https://quanteda.io for tutorials and examples.
## 
## Attaching package: 'quanteda'
## The following object is masked from 'package:utils':
## 
##     View
library("quanteda.dictionaries")

data_corpus_like <- corpus(like2)
data_likedfm <- dfm(data_corpus_like, dictionary = data_dictionary_MFD)
data_like <- dfm_subset(data_likedfm, ntoken(data_likedfm) > 0)
head(data_like)
## Document-feature matrix of: 6 documents, 10 features (88.3% sparse).
## 6 x 10 sparse Matrix of class "dfm"
##        features
## docs    care.virtue care.vice fairness.virtue fairness.vice loyalty.virtue
##   text2           0         0               1             0              0
##   text3           1         0               0             0              0
##   text4           0         0               0             0              1
##   text5           0         0               0             0              0
##   text6           0         0               0             0              0
##   text7           0         0               0             0              1
##        features
## docs    loyalty.vice authority.virtue authority.vice sanctity.virtue
##   text2            0                2              0               0
##   text3            0                0              0               0
##   text4            0                0              0               0
##   text5            0                0              0               1
##   text6            0                2              0               0
##   text7            0                0              0               0
##        features
## docs    sanctity.vice
##   text2             0
##   text3             0
##   text4             0
##   text5             0
##   text6             0
##   text7             0


# LIWCalike analysis

data_liwcalike_MFD <- liwcalike(data_corpus_like, dictionary = data_dictionary_MFD)
## Error in `[[<-.data.frame`(`*tmp*`, "WPS", value = c(16, 26, 6, 6.33333333333333, : replacement has 38778 rows, data has 38782
```

**Comment by kbenoit** [comment]
Fixed in v0.22. It was caused by removing punctuation (the default) from documents that were entirely punctuation, which removed the document entirely. Some of your documents were:
```
                     text2311                     text11524                     text32004                     text38300 
       "////////////////////" "///////////////////////////"                         "..."                           "." 
```

If you re-install and try it now, it should work.

**Comment by lin-jennifer** [comment]
Yes, it works now! Thank you so much for your help!
